src/processing_provider/QgepSwmm.py

```diff
@@ -0,0 +1,504 @@
+        con.commit()
+        return
+
+    def get_swmm_table(self, table_name):
+        """
+        Extract data from the swmm views in the database
+
+        Parameters:
+        table_name (string): Name of the view or tablle
+
+        Returns:
+        dic: table content
+        array: table attributes
+
+        """
+
+        # Connects to service and get data and attributes from tableName
+        con = psycopg2.connect(service=self.service)
+        cur = con.cursor()
+        try:
+            cur.execute('select * from qgep_swmm.{table_name}'.format(table_name=table_name))
+        except psycopg2.ProgrammingError:
+            print('Table %s doesnt exists' % table_name)
+            return None, None
+        data = cur.fetchall()
+        attributes = [desc[0] for desc in cur.description]
+
+        return data, attributes
+
+    def swmm_table(self, table_name):
+        """
+        Create swmm table
+
+        Parameters:
+        table_name (string): Name of the swmm section
+
+        Returns:
+        String: table content
+
+        """
+
+        # Create commented line which contains the field names
+        fields = ""
+        data, attributes = self.get_swmm_table(table_name)
+        if data is not None:
+            for i, field in enumerate(attributes):
+                # Does not write values stored in columns descriptions, tags and geom
+                if field not in ('description', 'tag', 'geom'):
+                    fields += field + "\t"
+
+            # Create input paragraph
+            tbl = u'[' + table_name + ']\n'\
+                ';;' + fields + '\n'
+            for feature in data:
+                for i, v in enumerate(feature):
+                    # Write description
+                    if attributes[i] == 'description' and str(v) != 'None':
+                        tbl += ';'
+                        tbl += str(v)
+                        tbl += '\n'
+
+                for i, v in enumerate(feature):
+                    # Does not write values stored in columns descriptions, tags and geom
+                    if attributes[i] not in ('description', 'tag', 'geom'):
+                        if str(v) != 'None':
+                            tbl += str(v) + '\t'
+                        else:
+                            tbl += '\t'
+                tbl += '\n'
+            tbl += '\n'
+            return tbl
+        else:
+            return '\n'
+
+    def copy_parameters_from_template(self, parameter_name):
+        """
+        Create swmm table from the template
+
+        Parameters:
+        parameter_name (string): Name of the swmm section to be copied
+
+        Returns:
+        String: section content
+
+        """
+        # Read template
+        options_template = open(self.options_template_file, 'r').read()
+        # Find and extract options
+        index_start = options_template.find('[%s]' % parameter_name)
```

Comment:
```suggestion
        index_start = options_template.find('[{parameter_name}]'.format(parameter_name=parameter_name))
```

This is more of a "modern python" recommendation. Use `format` instead of `%` formatting, it's more flexible and readable.